evalml/automl/automl_algorithm/iterative_algorithm.py

```diff
@@ -432,9 +435,14 @@ def _transform_parameters(self, pipeline, proposed_parameters):
                 component_parameters["n_jobs"] = self.n_jobs
             if "number_features" in init_params:
                 component_parameters["number_features"] = self.number_features
+            names_to_check = [
```

Comment:
This is https://github.com/alteryx/evalml/issues/3150

Reply:
question for my understanding: why do we need to check only specific components for `_pipeline_params` and not every component that shows up? [This is how it looks like in `DefaultAlgorithm`](https://github.com/alteryx/evalml/blob/main/evalml/automl/automl_algorithm/default_algorithm.py#L419-L421).

Reply:
I think this is precisely the tech debt around the #3150. Our current approach for specifying pipeline parameters through the search is really confusing.

My guess of what happened:
* In https://github.com/alteryx/evalml/pull/1676, we added the ability to freeze parameters/hyperparameters via the `pipeline_parameters` component. This had a [check for batch 0](https://github.com/alteryx/evalml/commit/51cba3fa041f8c94b60261e6c2ceb0f944b87ea5), to make sure we initialized the pipelines with values that are in the hyperparameter ranges. This could only apply for batch 0 otherwise we would always override the values proposed by the tuner with random values. 
* Then in https://github.com/alteryx/evalml/pull/2138, we added support to drop index columns throughout search. Since we can't specify the parameters for DropColumns as a hyperparameter, we had to manually add logic to add them.
* In https://github.com/alteryx/evalml/pull/2317, we separated the custom hyperparameters from pipeline parameters. The design choice was for pipeline parameters to only apply for the first batch and for custom hyperparemeters to apply for all subsequent batches. Therefore we're in a position were the only components whose parameters should be overwritten with what's in `names_to_check` are the select/drop columns transformers.

Regarding the difference in behavior with `DefaultAlgorithm`, that may be a bug? I think pipeline parameters should only apply in the first batch - at least that's what the design stipulated when we separated parameters from custom hyperparameters.

Reply:
Thanks for the context @freddyaboulton, really helpful in catching up on this logic. I made the explicit choice to pass down `pipeline_parameters` into each `DefaultAlgorithm` pipeline because my thought was that `pipeline_params` represented static parameters that a user wanted in each pipeline. But I think you're right that the design in #2317 [made it only apply in the first batch.](https://github.com/alteryx/evalml/pull/2317/files#r645831953). I filed #3153 to track this divergent behavior for further discussion!